webapp/src/Service/ScoreboardService.php

```diff
@@ -344,6 +351,28 @@ public function calculateScoreRow(
                 $judging = $submission->getJudgings()->first() ?: null;
             }
 
+            // three things will happen in the loop in this order:
+            // 1. update fastest runtime
+            // 2. count submissions until correct submission
+            // 3. determine time of first correct submission
+
+            // STEP 1:
+            // runtime improvements should be possible for all correct submissions
+            if (!is_null($judging) && $judging->getResult() == Judging::RESULT_CORRECT) {
+                $runtime = (int) round(1000*$judging->getMaxRuntime()); // round to milliseconds
```

Comment:
Nitpick: I think the correct thing to do here is to `floor`, not `round`. This is in general how time is dealt with. A clock showing hours and minutes doesn't display the next minute when it's past 30 seconds into that minute.